src/protocols/secure_channel/CASESession.cpp

```diff
@@ -684,9 +686,11 @@ CHIP_ERROR CASESession::HandleSigma2Resume(System::PacketBufferHandle && msg)
     mCASESessionEstablished = true;
 
     // Forget our exchange, as no additional messages are expected from the peer
+    mExchangeCtxt->SetDelegate(nullptr);
     mExchangeCtxt = nullptr;
```

Comment:
```suggestion
    // Make sure the exchange doesn't try to notify us when it closes,
    // since we might be dead by then.
    mExchangeCtxt->SetDelegate(nullptr);
    // Forget our exchange, as no additional messages are expected from the peer
    mExchangeCtxt = nullptr;
```
And similar at other callsites.

Reply:
I wonder whether we should factor out this two-call sequence, with the comments, into a helper method...

Reply:
Sure, that makes sense.

Will add that.

Reply:
> Approving, but let's do a followup to consolidate the must-be-done-just-so code.

I pushed up the helper, but am deferring work to move mExchangeContext into PairingSession.